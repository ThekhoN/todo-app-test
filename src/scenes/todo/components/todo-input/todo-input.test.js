import React from "react";
import TodoInput from "./index";
import { shallow } from "enzyme";

describe("TodoInput", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TodoInput />);
  });

  // start with this
  it("renders an input with class todo-input", () => {
    expect(
      wrapper.contains(
        <input
          type="text"
          className="todo-input"
          autoFocus
          placeholder="do something..."
        />
      )
    ).toEqual(true);
  });

  // snapshot
  it("snapshots correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // input.value === ""
  it("empty text value by default", () => {
    expect(wrapper.text()).toEqual("");
  });
});
