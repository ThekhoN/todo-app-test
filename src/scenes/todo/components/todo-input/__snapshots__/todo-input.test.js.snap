// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput snapshots correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "autoFocus": true,
      "className": "todo-input",
      "placeholder": "do something...",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoFocus": true,
        "className": "todo-input",
        "placeholder": "do something...",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
